docs/defining_effects.md

```diff
@@ -15,54 +15,45 @@ Effect types must have two type parameters: `m`, denoting any computations which
 Next, we can flesh out the definition of the `Teletype` effect by providing constructors for each primitive operation:
 
 ```haskell
-data Teletype m k
-  = Read (String -> m k)
-  | Write String (m k)
-  deriving (Functor)
+data Teletype (m :: Type -> Type) k where
+  Read  ::           Teletype m String
+  Write :: String -> Teletype m ()
```

Comment:
One consequence of this change is that first-order effects in general require kind annotations for the `m` parameter. (This is documented in the common errors document.)